tests/conformance_dsl/fragment.rs

```diff
@@ -391,11 +392,24 @@ impl WriteAsIon for ProxyElement<'_> {
                                     .map(|p| p.accepts_rest())
                                     .unwrap_or(false);
 
-                                // If this argument isn't in 'rest' position or it's already an
+                                // If this argument isn't in 'rest' position and it's not an
                                 // expression group...
-                                if !is_rest_parameter || arg_is_expr_group {
+                                if !is_rest_parameter && !arg_is_expr_group {
                                     // ...then wrap it in a `ProxyElement` and write it out.
                                     eexp_writer.write(ProxyElement(arg, self.1))?;
+                                } else if arg_is_expr_group {
+                                    // ...if it is an expression group, we need to encode it in an
+                                    // expression group built via our expression writer
+                                    // ProxyElement cannot do this alone.
+                                    let mut group_writer = eexp_writer.expr_group_writer()?;
+                                    let group_args = arg
+                                        .as_sexp()
+                                        .unwrap() // Verified above
+                                        .iter()
+                                        .skip(1);
```

Comment:
This is skipping the first element, the `#$::`, right? If I am correct could you add a comment for posterity, and if I am wrong could you add a comment telling me what is the deal here?

Reply:
Yep that's exactly why. I'll add a comment, good call.